contracts/schemes/ContributionRewardExt.sol

```diff
@@ -0,0 +1,454 @@
+    }
+
+    mapping(bytes32=>ContributionProposal) public organizationProposals;
+
+    IntVoteInterface public votingMachine;
+    bytes32 public voteParams;
+    Avatar public avatar;
+    address public rewarder;
+
+    /**
+    * @dev enables this contract to receive ethers
+    */
+    function() external payable {
+    }
+
+    /**
+     * @dev initialize
+     * @param _avatar the avatar to mint reputation from
+     * @param _votingMachine the voting machines address to
+     * @param _voteParams voting machine parameters.
+     * @param _rewarder an address which allowed to redeem the contribution.
+       if _rewarder is 0 this param is agnored.
+     */
+    function initialize(
+        Avatar _avatar,
+        IntVoteInterface _votingMachine,
+        bytes32 _voteParams,
+        address _rewarder
+    )
+    external
+    {
+        require(avatar == Avatar(0), "can be called only one time");
+        require(_avatar != Avatar(0), "avatar cannot be zero");
+        avatar = _avatar;
+        votingMachine = _votingMachine;
+        voteParams = _voteParams;
+        rewarder = _rewarder;
+    }
+
+    /**
+    * @dev execution of proposals, can only be called by the voting machine in which the vote is held.
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _decision a parameter of the voting result, 1 yes and 2 is no.
+    */
+    function executeProposal(bytes32 _proposalId, int256 _decision)
+    external
+    onlyVotingMachine(_proposalId)
+    returns(bool) {
+        require(organizationProposals[_proposalId].executionTime == 0);
+        require(organizationProposals[_proposalId].beneficiary != address(0));
+        // Check if vote was successful:
+        if (_decision == 1) {
+          // solhint-disable-next-line not-rely-on-time
+            organizationProposals[_proposalId].executionTime = now;
+        }
+        emit ProposalExecuted(address(avatar), _proposalId, _decision);
+        return true;
+    }
+
+    /**
+    * @dev Submit a proposal for a reward for a contribution:
+    * @param _descriptionHash A hash of the proposal's description
+    * @param _reputationChange - Amount of reputation change requested .Can be negative.
+    * @param _rewards rewards array:
+    *         rewards[0] - Amount of tokens requested per period
+    *         rewards[1] - Amount of ETH requested per period
+    *         rewards[2] - Amount of external tokens requested per period
+    * @param _externalToken Address of external token, if reward is requested there
+    * @param _beneficiary Who gets the rewards. if equal to 0 the beneficiary will be msg.sender.
+    * @param _proposer proposer . if equal to 0 the proposer will be msg.sender.
+    */
+    function proposeContributionReward(
+        string memory _descriptionHash,
+        int256 _reputationChange,
+        uint[3] memory _rewards,
+        IERC20 _externalToken,
+        address payable _beneficiary,
+        address _proposer
```

Comment:
Why would you want to allow anyone to propose in the name of others?

Reply:
1. because we need that to the competition .
2. to allow propose on behalf of someone else...e.g to get rep from hc ..to this address